simulation/test_torrent_status.cpp

```diff
@@ -41,62 +41,139 @@ using namespace sim;
-				start_time = lt::clock_type::now();
+				lt::time_point start_time = lt::clock_type::now();
 				handle = ta->handle;
+				torrent_status st = handle.status();
+				// test last upload and download state before wo go throgh
+				// torrent states
+				TEST_CHECK(st.last_download == start_time);
+				TEST_CHECK(st.last_upload == start_time);
 			}
 		}
 		// terminate
 		, [&](int ticks, lt::session&) -> bool
 		{
-
+			last_tick = ticks;
 			// simulate 20 hours of uptime. Currently, the session_time and internal
 			// peer timestamps are 16 bits counting seconds, so they can only
 			// represent about 18 hours. The clock steps forward in 4 hour increments
 			// to stay within that range
 			if (ticks > 20 * 60 * 60)
+			{
 				return true;
+			}
 
-			// once an hour, verify that the timers seem correct
-			if ((ticks % 3600) == 0)
+			if(tick_is_in_active_range){
+				// 1 second per tick
+				active_time++;
+			}
+
+			// resume on a tick which is far away from next tick check intervall
+			// because with very low upload limit it takes some time to unchock
+			switch(ticks)
+			{
+				case 0:
+					// torrent get ready for seeding on first tick, means time +1s
+					startseed_time = lt::clock_type::now();
+					tick_is_in_active_range = true;
+					break;
+				case 21:
+					handle.pause();
+					tick_is_in_active_range = false;
+					last_active_time = lt::clock_type::now();
+					break;
+				case 64001:
+					handle.resume();
+					tick_is_in_active_range = true;
+					break;
+				case 64061:
+					handle.pause();
+					tick_is_in_active_range = false;
+					last_active_time = lt::clock_type::now();
+					break;
+				case 70001:
+					handle.resume();
+					tick_is_in_active_range = true;
+					break;
+				case 70061:
+					// wait for 72000 or upload finish and check only sometimes
+					ticks_check_intervall = 1000;
+					break;
+			}
+			// verify that the timers seem correct
+			if (tick_is_in_active_range && (ticks % ticks_check_intervall) == 0)
 			{
 				lt::time_point now = lt::clock_type::now();
-				auto const since_start = duration_cast<seconds>(now - start_time) - lt::seconds(1);
 				torrent_status st = handle.status();
-				TEST_CHECK(st.active_duration == since_start);
-				TEST_CHECK(st.seeding_duration == since_start);
-				TEST_CHECK(st.finished_duration == since_start);
-				TEST_CHECK(st.last_upload < start_time);
-
-				// checking the torrent counts as downloading
-				// eventually though, we've forgotten about it and go back to -1
-				if (since_start > lt::seconds(65000))
-				{
-					TEST_CHECK(st.last_download < start_time);
-				}
-				else
-				{
-					// TODO: this should really be a proximity-check
-					TEST_CHECK(st.last_download == start_time + lt::seconds(1));
-				}
+				TEST_CHECK(st.active_duration == active_time);
+				TEST_CHECK(st.seeding_duration == active_time);
+				TEST_CHECK(st.finished_duration == active_time);
+				printf("active_time: %ld \n", active_time.count());
```

Comment:
@arvidn I need to remove the printf anyway beause on windows %ld does not work because it's not a long its int64, weird.
If count is safe to use I could write TEST_EQUAL(st.active_duration.count(), active_time.count()); otherwise we need to make something to make TEST_EQUAL compatible with durations.